src/storage/secondary/rowset/disk_rowset.rs

```diff
@@ -131,10 +135,52 @@ impl DiskRowset {
             .sum1()
             .unwrap_or(0)
     }
+
+    /// Get the start row id to begin with for later table scanning.
+    /// If `begin_keys` is empty, we return `ColumnSeekPosition::RowId(0)` to indicate scanning
+    /// from the beginning, otherwise we scan the rowsets' first column indexes, find the first
+    /// block who contains data greater than or equal to `begin_key` and return the row id of
+    /// the block's first key. Currently, only the first column of the rowsets can be used to get
+    /// the start row id and this column should be primary key.
+    /// If `begin_key` is greater than all blocks' `first_key`, we return the `first_key` of the
+    /// last block.
+    /// Todo: support multi sort-keys range filter
+    pub async fn start_rowid(&self, begin_keys: &[DataValue]) -> ColumnSeekPosition {
+        if begin_keys.is_empty() {
+            return ColumnSeekPosition::RowId(0);
+        }
+
+        // for now, we only use the first column to get the start row id, which means the length
+        // of `begin_keys` can only be 0 or 1.
+        let begin_key = begin_keys[0].borrow();
+        let column = self.column(0);
+        let column_index = column.index();
+
+        let start_row_id = match *begin_key {
+            DataValue::Int32(begin_val) => {
+                let mut pre_block_first_key = 0;
+                for index in column_index.indexes() {
+                    let mut first_key: &[u8] = &index.first_key;
+                    let first_val: i32 = PrimitiveFixedWidthEncode::decode(&mut first_key);
+
+                    if first_val > begin_val {
+                        break;
+                    }
+                    pre_block_first_key = index.first_rowid;
+                }
+                pre_block_first_key
+            }
+            // Todo: support ohter type
+            _ => panic!("for now support range-filter scan by sort key type of int32"),
+        };
+        ColumnSeekPosition::RowId(start_row_id)
```

Comment:
This positioning is not accurate. If we have 2 blocks with start of:

```
block 1: a
block 2: d
```

Now we want to seek to `b`. This function will return row_id = 0. However, the first row from block 1 cannot be `b` -- it's `a` instead. Therefore, `RowSetIterator` will still need `begin_sort_key` as parameter, and it will need to filter those keys below start sort key.

Reply:
oh .the line 166 should be >= instead of >

Reply:
I mean the begin row might not be the same as sort key? In this case, the RowSetIterator might produce more items than expected. You may take a look at the above comment, and put range filters into the filter condition when unit testing.

Reply:
oh yes. i forget this